core/src/main/java/org/kiwix/kiwixmobile/core/search/DefaultAdapter.java

```diff
@@ -0,0 +1,102 @@
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.search;
+
+import android.content.Context;
+import android.text.Html;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.TextView;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.recyclerview.widget.RecyclerView;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import org.kiwix.kiwixmobile.core.R;
+
+import static javax.swing.UIManager.getString;
+import static org.kiwix.kiwixmobile.core.utils.StyleUtils.dialogStyle;
+
+public class DefaultAdapter extends RecyclerView.Adapter<DefaultAdapter.ViewHolder> {
+
+  private Context context;
+  private AdapterView.OnItemClickListener onItemClickListener;
+  List<String> searchList;
+  @Inject
+  SearchPresenter searchPresenter;
+
+  public DefaultAdapter(Context context) {
+    this.context = context;
+  }
+
+  @NonNull @Override
+  public DefaultAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+    View view = (View) LayoutInflater.from(parent.getContext())
+      .inflate(android.R.layout.simple_list_item_1, parent, false);
+    ViewHolder holder = new ViewHolder(view);
+    return holder;
+  }
+
+  @Override public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+    holder.recentSearch.setText(
+      Html.fromHtml(getItem(position)).toString());///need to get rid of this error
+  }
+
+  @Override public int getItemCount() {
+    return searchList.size();
+  }
+
+  @NonNull public String getItem(int position) {
+    return searchList.get(position);
+  }
+
+  public class ViewHolder extends RecyclerView.ViewHolder {
+    @Nonnull
+    TextView recentSearch;
+
+    @Nonnull
+    public ViewHolder(@NonNull View itemView) {
+      super(itemView);
+      recentSearch = itemView.findViewById(android.R.id.text1);
+      itemView.setOnClickListener(new View.OnClickListener() {
+        @Override public void onClick(View v) {
+          if (context instanceof SearchActivity) {
+            String title = recentSearch.getText().toString();
+            ((SearchActivity) context).searchPresenter.saveSearch(title);
```

Comment:
@macgills This is what I was talking about. If this is not the ideal way as you said, I don't know how should I go about it.I didn't get what you said.

Reply:
```
class DefaultAdapter{
interface ClickListener{
void onItemClick(String text);
void onItemLongClick(String text);
}
private final ClickListener clickListener;
public DefaultAdapter(ClickListener clickListener){
    this.clickListener = clickListener
  }
.
.
.
itemView.setOnClickListener(v-> clickListener.onItemClick(recentSearch.getText().toString()))
}
```

abridged but how you do it

Reply:
can you please guide me through it. I have not done it so far in my life

Reply:
the only really missing piece is the code to write from the calling location, here an activity but this concept of callbacks is very generalised. There are further examples [here](https://stackoverflow.com/questions/18279302/how-do-i-perform-a-java-callback-between-classes), perhaps if you try asking me very specific questions I can try and shed some light

Reply:
I think I have solved the issue. Please check it out.